src/compiler/parser.ts

```diff
@@ -3337,6 +3347,7 @@ namespace Parser {
             case ParsingContext.JsxAttributes: return parseErrorAtCurrentToken(Diagnostics.Identifier_expected);
             case ParsingContext.JsxChildren: return parseErrorAtCurrentToken(Diagnostics.Identifier_expected);
             case ParsingContext.AssertEntries: return parseErrorAtCurrentToken(Diagnostics.Identifier_or_string_literal_expected); // AssertionKey.
+            case ParsingContext.JSDocComment: return parseErrorAtCurrentToken(Diagnostics.Identifier_expected);

```

Comment:
What does it mean to provide an error in a JSDoc parsing comment? Where would we issue this?

Reply:
This is part of recovery in parseList/parseDelimitedList and feasibly only only happens when a list ends prematurely.

Reply:
You probably don't need an error but it's fine. In a follow-up PR, I'd be curious to see what happens if it `Debug.fail()`s.

Reply:
Huh, yeah, you can just `Debug.fail()` and this PR also works. Maybe I'll just do that.

But, I am probably close to eliminating this entire parse mode entirely too.

Reply:
Probably this doesn't break because the tests aren't good enough.